webgateway_json_api.md

```diff
@@ -0,0 +1,392 @@
+
+
+ * [API Overview](#api-overview)
+    * [Omero-marshal and Projection-based APIs](#omero-marshal-and-projection-based-apis)
+    * [Versioning](#versioning)
+    * [Pagination](#pagination)
+    * [Normalizing Experimenters and Groups](#normalizing-experimenters-and-groups)
+    * [Child Counts](#child-counts)
+    * [Error handling](#error-handling)
+ * [Image containers](#image-containers)
+    * [List projects](#list-projects)
+    * [Create a project](#create-a-project)
+    * [Get a single project](#create-a-single-project)
+    * [Delete a project](#delete-a-project)
+
+
+API Overview
+============
+
+The OMERO json API described here provides create, read, update and delete
+access to an underlying OMERO server.
+
+
+Omero-marshal and Projection-based APIs
+---------------------------------------
+The majority of the API urls use [omero-marshal](https://github.com/openmicroscopy/omero-marshal)
+to generate json dictionaries from OMERO **model** objects. The json dictionaries will be populated
+with all fields of the OMERO model object, such that changing a single field and sending this
+back to OMERO will update the object accordingly.
+All these url are under the ```m``` prefix:
+
+    <server>/webgateway/api/v1.0/m/
```

Comment:
A few secondary thoughts on the versioning scheme. What is the meaning of `x.y`  in terms of version bumps and API compatibility?

The issue with exposing multiple numbers in the URL is that it might forces consumers to change the URLs everytime a new non-breaking change is released which might not be desirable. Back to a previous discussion, we could decide to reduce this to the `MAJOR` version number as in `vMAJOR`. However this means we need to be strict with regard to compatibility.

As expressed in the various calls, our current effort is to work on is `0.1.0` according to the semver semantics, i.e. I do not think we want to expose this as `v1*` but rather `v0` or `v0.1` depending on the policy above.